cmd/kubeadm/app/util/marshal.go

```diff
@@ -66,67 +69,74 @@ func UnmarshalFromYamlForCodecs(buffer []byte, gv schema.GroupVersion, codecs se
 	return runtime.Decode(decoder, buffer)
 }
 
-// ExtractAPIVersionAndKindFromYAML extracts the APIVersion and Kind fields from YAML bytes
-func ExtractAPIVersionAndKindFromYAML(b []byte) (string, string, error) {
-	decoded, err := LoadYAML(b)
-	if err != nil {
-		return "", "", fmt.Errorf("unable to decode config from bytes: %v", err)
-	}
-
-	kindStr, ok := decoded["kind"].(string)
-	if !ok || len(kindStr) == 0 {
-		return "", "", fmt.Errorf("any config file must have the kind field set")
-	}
-	apiVersionStr, ok := decoded["apiVersion"].(string)
-	if !ok || len(apiVersionStr) == 0 {
-		return "", "", fmt.Errorf("any config file must have the apiVersion field set")
+// SplitYAMLDocuments reads the YAML bytes per-document, unmarshals the TypeMeta information from each document
+// and returns a map between the GroupVersionKind of the document and the document bytes
+func SplitYAMLDocuments(yamlBytes []byte) (map[schema.GroupVersionKind][]byte, error) {
```

Comment:
Please fail if there two objects of the same kind

Reply:
fixed